### DISCUSSION TITLE
Monitoring tool

### DISCUSSION THREAD

**Discussion by mochiruntime** [clicepfl/website#12] [Ideas]
At some point we'll probably want to monitor the health of our services. Here are a few tools we could use:

- [Sentry](https://sentry.io): it is awesome, easy to setup and has a ton of integrations. It's open-source but available as a cloud service, though self-hosting is strongly discouraged as the steup is rather heavy and cumbersome. The upside of using Sentry SaaS is that it can detect and report outages and problems even if our services are down (e.g. the server crashes, problems with the reverse proxy, ...).

- [Glitchtip](https://glitchtip.com/): lightweight alternative which can be self-hosted. It can also provide uptime monitoring, though since we would host it on the same machine, its purpose would be limited to application-level outages.

**Comment by Lumycea** [comment]
It would greatly depend on the orchestration tool we use. As we will use containers for all services, we could go for:
 - [Kubernetes](https://kubernetes.io/) which has built-in monitoring tools
 - [Docker swarm](https://docs.docker.com/engine/swarm/). It has no built-in tools but we could add one (like [Prometheus](https://prometheus.io/docs/guides/dockerswarm/))

**Reply by mochiruntime** [reply]
Yes, and we'll probably also have them depending on the distribution!
Though the tools I mentioned are concerned about internal metrics, so we can integrate stuff like tracing, profiling, custom measurements, ... :blush: The cool thing about Sentry (and Glitchtip since it is API-compliant with Sentry) is that it's very easy to integrate, by simply attaching to lifecycle events. It also has out of the box support for Next.js and Strapi also developed a plugin :+1:
